src/waveform/renderers/waveformrendererpreroll.cpp

```diff
@@ -44,8 +44,8 @@ void WaveformRendererPreroll::draw(QPainter* painter, QPaintEvent* event) {
         double samplesPerPixel = m_waveformRenderer->getVisualSamplePerPixel();
         double numberOfSamples = m_waveformRenderer->getLength() * samplesPerPixel;
 
-        int currentPosition = m_waveformRenderer->getPlayPosVSample();
-        int totalSamples = m_waveformRenderer->getTotalVSample();
+        int currentPosition = static_cast<int>(m_waveformRenderer->getPlayPosVSample());
+        int totalSamples = static_cast<int>(m_waveformRenderer->getTotalVSample());
```

Comment:
Ups sorry. you have hit a rats nest. 
getPlayPosVSample() should directly return int, because it is the already rounded VSample. 
However I think we have some rounding issues here, because m_visualSamplePerPixel is double and can be changed in fine steps since we have removed the stepping from waveform zooms. 
I think we need to review all position calculations.

Reply:
Done, see #3176.